### PR TITLE
[New Term] C++ Math Functions: `exp2()`

### LINKED ISSUE #962: [Term Entry] C++ Math Functions: `exp2()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `exp2()` function.

- In the `terms/` folder, make a new folder called `exp2`
- Inside the `exp2` folder, create a new markdown file called `exp2.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by itsTodFod**
## Description

added new term entry C++ Math Functions: `exp2()`

Closes #962

## Type of Change

Adding a new entry

## Checklist
* [x]   All writings are my own.
* [x]    My entry follows the Codecademy Docs style guide.
* [x]    My changes generate no new warnings.
* [x]    I have performed a self-review of my own writing and code.
* [x]    I have checked my entry and corrected any misspellings.
* [x]    I have made corresponding changes to the documentation if needed.
* [x]    I have confirmed my changes are not being pushed from my forked main branch.
* [x]    I have confirmed that I'm pushing from a new branch named after the changes I'm making.
* [x]    Under "Development" on the right, I have linked any issues that are relevant to this PR.

### CODE REVIEW

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Functions'
+  - 'Arithmetic'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The **`exp2()`** is a builtin function in C++ STL that computes the `base-2` exponential function of a given number, i.e. `2` raised to the given argument.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/exp2/exp2.md:14]:
```suggestion
The **`exp2()`** is a built-in function in C++ STL that computes the `base-2` exponential function of a given number, i.e. `2` raised to the given argument.
```

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+## Syntax
+
+```pseudo
+exp2(x)
+```
+
+It takes a single mandatory argument (can be positive, negative or 0).
+The `exp2()` function returns the value in the range of `[0, ∞]`.
+
+If no errors occur, the base-2 exponential of `n` is returned.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/exp2/exp2.md:25]: These lines, in addition to the "Error Handling" lines should all be in a single bulleted list.

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+<details>
+	<summary>Error Handling</summary>
+
+	If the argument is ±0, 1 is returned.
+	If the argument is -∞, +0 is returned.
+	If the argument is +∞, +∞ is returned.
+	If the argument is NaN, NaN is returned.
+</details>
+
+The return type for the `exp2()` function is `double`, `float`, `long double`.
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/exp2/exp2.md:37]:
```suggestion
The return type for the `exp2()` function is `double`, `float`, or `long double`.
```

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+---
+Title: 'exp2()'
+Description: 'Returns the base-2 exponential function, i.e. 2 raised to the given argument'
```
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/exp2/exp2.md:3]:
```suggestion
Description: 'Returns the base-2 exponential function, i.e. 2 raised to the given argument.'
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@itsTodFod I did a first review and left a few comments.

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md (hunk 2 above)
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/exp2/exp2.md]: Please check if the list is as expected.

**Comment by itsTodFod** [comment]
Hi @SSwiniarski 
Committed changes as asked, please review.

**Review by SSwiniarski — APPROVED** [review]
@itsTodFod Looks good for second review

File: content/cpp/concepts/math-functions/terms/exp2/exp2.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+ - It takes a single mandatory argument (can be positive, negative or 0).
+ - The `exp2()` function returns the value in the range of `[0, ∞]`.
+ - If no errors occur, the base-2 exponential of `n` is returned.
+ - If a range error due to overflow occurs, `HUGE_VAL`, `HUGE_VALF`, or `HUGE_VALL` is returned.
+ - If a range error occurs due to underflow, the correct result (after rounding) is returned.
+ - If the argument is ±0, 1 is returned.
+ - If the argument is -∞, +0 is returned.
+ - If the argument is +∞, +∞ is returned.
+ - If the argument is NaN, NaN is returned.
+ - The return type for the `exp2()` function is `double`, `float`, or `long double`.
```
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/exp2/exp2.md:31]:
```suggestion
 - If a range error due to overflow occurs, `HUGE_VAL`, `HUGE_VALF`, or `HUGE_VALL` is returned.
 - If a range error occurs due to underflow, the correct result (after rounding) is returned.
 - If the argument is ±0, 1 is returned.
 - If the argument is -∞, +0 is returned.
 - If the argument is +∞, +∞ is returned.
 - If the argument is `NaN`, `NaN` is returned.
 - The return type for the `exp2()` function is `double`, `float`, or `long double`.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@itsTodFod Left a few comments/suggestions 😄

**Comment by Dusch4593** [comment]
Hey @itsTodFod just wanted to check in and see if you had any questions or needed any help with implementing the most recent feedback?

**Comment by SSwiniarski** [comment]
@itsTodFod, because it's been over two weeks of inactivity on this PR, I went ahead and committed @Dusch4593's suggestions in the interest of getting this merged.

@Dusch4593, can you check for anything else so we can approve and merge this?

**Review by Dusch4593 — APPROVED** [review]
@itsTodFod This looks good for merge. 👍🏻

I made a few changes in 0891fcfbe5c4955552d39430bb368087421ce5c8 and 7d37e99ce1eb85291cc0223db74754bb50b7d781 that I wanted to note:

- Edits to opening sentence on line # 14.
- Removed leading spaces from list in "Syntax" section.
- Placed a trailing empty line after line # 77 (per lint configurations).

**Comment by Dusch4593** [comment]
Congrats and thank you for contributing to Codecademy Docs, @itsTodFod! 🚀 🥳 🎊

The entry should now be live: https://www.codecademy.com/resources/docs/cpp/math-functions/exp2

Your Codecademy profile is currently anonymously listed on the entry page. To make it visible, you'll need to add your GitHub username to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account:

Screen Shot 2022-11-21 at 9 16 29 AM

Looking forward to your next Docs contribution! 😄

**Pull request merged**